# HG changeset patch
# User Ryan Hinton
# Date 1203929793 18000
#      Mon Feb 25 03:56:33 2008 -0500
# Branch release-3-0-x
# Node ID 6adc43cce4c6fb3211653748e0a7cd23470fd098
# Parent  803bb6fdbad5377413fadd545508fac2e2c94ddd
unpack.m: pass "-f -" to tar

diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -59,16 +59,17 @@ Etienne Grossmann
 Peter Gustafson
 Kai Habel
 William P. Y. Hadisoeseno
 Benjamin Hall
 Kim Hansen
 Soren Hauberg
 Daniel Heiserer
 Yozo Hida
+Ryan Hinton
 Roman Hodek
 A. Scottedward Hodel
 Richard Allan Holcombe
 Tom Holroyd
 David Hoover
 Kurt Hornik
 Christopher Hulbert
 Cyril Humbert
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-02-25  Ryan Hinton  <rwh4s@virginia.edu>
+
+	* miscellaneous/unpack.m: Use "-f -" args for tar.
+
 2008-02-22  John W. Eaton  <jwe@octave.org>
 
 	* miscellaneous/fullfile.m: Improve handling of empty args and
 	args ending with filesep.
 
 	* control/base/__stepimp__.m: Don't use subplot for just one plot.
 
 2008-02-21  John W. Eaton  <jwe@octave.org>
diff --git a/scripts/miscellaneous/unpack.m b/scripts/miscellaneous/unpack.m
--- a/scripts/miscellaneous/unpack.m
+++ b/scripts/miscellaneous/unpack.m
@@ -119,22 +119,22 @@ function filelist = unpack (file, direct
     commandlist.z = commandlist.gz;
     commandlist.bz2 = {"bzip2 -d -v \"%s\"", ...
 		       "bzip2 -d \"%s\"", ...
 		       @__parse_bzip2__, true};
     commandlist.bz = commandlist.bz2;
     commandlist.tar = {"tar -x -v -f \"%s\"", ...
 		       "tar -x -f \"%s\"", ...
 		       @__parse_tar__, false};
-    commandlist.targz = {"gzip -d -c \"%s\" | tar -x -v", ...
-			 "gzip -d -c \"%s\" | tar -x", ...
+    commandlist.targz = {"gzip -d -c \"%s\" | tar -x -v -f -", ...
+			 "gzip -d -c \"%s\" | tar -x -f -", ...
 			 @__parse_tar__, false};
     commandlist.tgz = commandlist.targz;
-    commandlist.tarbz2 = {"bzip2 -d -c \"%s\" | tar -x -v", ...
-			  "bzip2 -d -c \"%s\" | tar -x", ...
+    commandlist.tarbz2 = {"bzip2 -d -c \"%s\" | tar -x -v -f -", ...
+			  "bzip2 -d -c \"%s\" | tar -x -f -", ...
 			  @__parse_tar__, false};
     commandlist.tarbz = commandlist.tarbz2;
     commandlist.tbz2 = commandlist.tarbz2;
     commandlist.tbz = commandlist.tarbz2;
     commandlist.zip = {"unzip \"%s\"", ...
 		       "unzip -q \"%s\"", ...
 		       @__parse_zip__, false};
   endif
